### PR TITLE
Openjpeg2

### PR DESCRIPTION
**Pull request by JPeterMugaas**

### CODE REVIEW

**Comment by Alexpux** [comment]
Update checksums

**Comment by Alexpux** [comment]
Don't see checksums update. You must do it yourself. Github will not do it for you

**Comment by JPeterMugaas** [comment]
I did run updpkgsums and copied the file to the openjpeg2 branch.  Even when I run updpkgsums here, I still get:
sha256sums=('4afc996cd5e0d16360d71c58216950bcb4ce29a3272360eb29cadb1c8bce4efc'
            '6d9d88474f815ddcd16b230c96aeb27171a640c5636e52182ae7cd1a1b5e23a3'
            '756f93b18ac80ea1cf90186d01800fcbe4dc40b36f164816f327f47b028902e1'
            'b7b22dca26d5fd98ca5bc6ce775bbbabebc0e10fcf07fd0afb54d580699b8312'
            'fcbc87fc5e51721e25ad9e50e84c336c9a5902ee8b2f2bb0800a3ea3fd72988e')

**Comment by mati865** [comment]
Because `source` is wrong:
```
source=(${_realname}-${pkgver}.tar.gz::https://github.com/uclouvain/openjpeg/archive/version.${pkgver%.*}.tar.gz
```
It downloads version 2.1 and renames it to 2.1.2.

**Comment by Alexpux** [comment]
Yeah, need change source url to get proper sources

**Comment by Alexpux** [comment]
Will it be fixed?

**Comment by JPeterMugaas** [comment]
I have fixed it locally unfortunately, the new version doesn't seem to build I'm not sure why.  It seems to happen when I try making a shared version of the library.  I get this:
[ 81%] Linking C executable ../../../bin/opj_decompress.exe
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text+0x1101): undefined reference to `opj_version'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x8e): undefined reference to `opj_set_default_decoder_parameters'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x174): undefined reference to `opj_stream_create_default_file_stream'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1dd): undefined reference to `opj_stream_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x27a): undefined reference to `opj_create_decompress'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x299): undefined reference to `opj_set_info_handler'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x2ab): undefined reference to `opj_set_warning_handler'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x2bd): undefined reference to `opj_set_error_handler'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x30a): undefined reference to `opj_setup_decoder'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x327): undefined reference to `opj_read_header'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x376): undefined reference to `opj_set_decode_area'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x393): undefined reference to `opj_decode'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x3a8): undefined reference to `opj_end_decompress'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x408): undefined reference to `opj_stream_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x67b): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x78f): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x848): undefined reference to `opj_get_decoded_tile'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x898): undefined reference to `opj_create_decompress'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x8af): undefined reference to `opj_create_decompress'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x914): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x940): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xa84): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xd70): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xe15): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xe22): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xe2f): undefined reference to `opj_destroy_cstr_index'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xfbc): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xfc9): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0xfd6): undefined reference to `opj_destroy_cstr_index'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x10fe): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x112a): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1167): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1171): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1191): undefined reference to `opj_stream_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x119b): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x11a8): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x11cd): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x11d5): undefined reference to `opj_stream_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x11e2): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1205): undefined reference to `opj_stream_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x120f): undefined reference to `opj_destroy_codec'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x1386): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(opj_decompress.c.obj):opj_decompress.c:(.text.startup+0x13a9): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x1746): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x1ab1): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x27a7): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x296a): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x3488): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x41a8): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x4363): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x51d9): undefined reference to `opj_version'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x54a2): undefined reference to `opj_version'
CMakeFiles/opj_decompress.dir/objects.a(convert.c.obj):convert.c:(.text+0x57b8): undefined reference to `opj_version'
CMakeFiles/opj_decompress.dir/objects.a(convertbmp.c.obj):convertbmp.c:(.text+0x12c1): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(convertbmp.c.obj):convertbmp.c:(.text+0x1374): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convertbmp.c.obj):convertbmp.c:(.text+0x14cc): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(converttif.c.obj):converttif.c:(.text+0x39ae): undefined reference to `opj_image_create'
CMakeFiles/opj_decompress.dir/objects.a(converttif.c.obj):converttif.c:(.text+0x3eaa): undefined reference to `opj_image_destroy'
CMakeFiles/opj_decompress.dir/objects.a(convertpng.c.obj):convertpng.c:(.text+0xacf): undefined reference to `opj_image_create'
collect2.exe: error: ld returned 1 exit status

I have commited what I have so someone else can look.

File: mingw-w64-openjpeg2/0006-openjpeg-2.1.0-stdcall-for-all-win.patch [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-@@ -76,6 +76,12 @@
- 	#define OPJ_DEPRECATED(func) func
- #endif
- 
+--- openjpeg-2.1.2/src/lib/openmj2/openjpeg.h.orig	2016-10-02 12:10:38.105864300 -0400
++++ openjpeg-2.1.2/src/lib/openmj2/openjpeg.h	2016-10-02 12:12:33.009046900 -0400
+@@ -44,6 +44,11 @@
+    Compiler directives
+ ==========================================================
+ */
```
Comment (Alexpux) [inline mingw-w64-openjpeg2/0006-openjpeg-2.1.0-stdcall-for-all-win.patch]: This open multiline comment that disable calling convention for compiler. Issue is here

**Comment by JPeterMugaas** [comment]
Still no joy because is definitely not copen and I still can not see what the problem with this actually is.

**Comment by Alexpux** [comment]
I think the problem is in calling convention

**Comment by JPeterMugaas** [comment]
I think it may have something to do with openjpip because I could build with that disabled.  I'm not sure what is going on here as that doesn't seem to have clear headers for the library.

**Comment by Biswa96** [comment]
@JPeterMugaas Can you remember why 0006-openjpeg-2.1.0-stdcall-for-all-win.patch was commented out? I my testing the patch fixes static linking with 32 bit library.

**Comment by JPeterMugaas** [comment]
I can't really remember and that was years ago.

**Pull request merged**